src/runtime/hexagon_remote/halide_remote.cpp

```diff
@@ -378,36 +378,35 @@ int halide_hexagon_remote_run_v2(handle_t module_ptr, handle_t function,
     // Get a pointer to the argv version of the pipeline.
     pipeline_argv_t pipeline = reinterpret_cast<pipeline_argv_t>(function);
 
-    // Construct a list of arguments. This is only part of a
-    // buffer_t. We know that the only field of buffer_t that the
-    // generated code should access is the host field (any other
-    // fields should be passed as their own scalar parameters) so we
-    // can just make this dummy buffer_t type.
-    struct buffer_t {
+    // Construct a list of arguments. This is the way that HexagonOffload
+    // packages arguments for us; despite the name, it really has nothing
+    // to do with the old 'buffer_t' type (or the current 'halide_buffer_t' type).
```

Comment:
This comment has a lot of "history" that is irrelevant to the current code. How about just calling this struct "hexagon_device_pointer" or something like that, and removing all references to "buffer_t"? Instead there should be a comment on the code that translates buffers to pointers (and the other args).

Reply:
PTAL